src/lib/containers/table/SynapseTable.tsx

```diff
@@ -791,20 +816,37 @@ export default class SynapseTable extends React.Component<
           return <td className="SRC-hidden" key={`(${rowIndex},${colIndex})`} />
         },
       )
+
+      const entityVersionNumber = row.versionNumber?.toString()
+      const rowSynapseId = `syn${row.rowId}`
+      const fileEntityHandle = this.state.fileEntityHandleArray ? this.state.fileEntityHandleArray[rowIndex] : undefined
+
       // also push the access column value if we are showing user access for individual items (still shown if not logged in)
       if (isShowingAccessColumn) {
-        const rowSynapseId = `syn${row.rowId}`
         rowContent.unshift(
           <td key={rowSynapseId} className="SRC_noBorderTop">
             <HasAccess
               key={rowSynapseId}
               entityId={rowSynapseId}
-              entityVersionNumber={row.versionNumber?.toString()}
+              entityVersionNumber={entityVersionNumber}
               token={token}
             ></HasAccess>
           </td>,
         )
       }
+
+      if (isShowingDownloadColumn) {
+        rowContent.unshift(
+          <td className="SRC_noBorderTop direct-download">
+            <DirectDownload
+              key={"direct-download-"+rowSynapseId}
+              token={token}
+              fileEntityHandle={fileEntityHandle}
+            ></DirectDownload>
+          </td>
```

Comment:
I wonder if I should move the icon switching concept outside the <DirectDownload> component. For example:

<DirectDownload
key={"direct-download-"+rowSynapseId}
token={sting}
isAuthorized={boolean}
downloadLink={string}
externalUrl={string}
></DirectDownload>

Reply:
It would also mean moving the logic to get the download link on click (since you can't pass these in, because the presigned urls expire).  I like what you've done